biodatasets/n2c2_2008/n2c2_2008.py

```diff
@@ -0,0 +1,366 @@
+            name="n2c2_2008_source",
+            version=SOURCE_VERSION,
+            description="n2c2_2008 source schema",
+            schema="source",
+            subset_id="n2c2_2008",
+        ),
+        BigBioConfig(
+            name="n2c2_2008_bigbio_text",
+            version=BIGBIO_VERSION,
+            description="n2c2_2008 BigBio schema",
+            schema="bigbio_text",
+            subset_id="n2c2_2008",
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = "n2c2_2008_source"
+
+    def _info(self) -> datasets.DatasetInfo:
+
+        if self.config.schema == "source":
+            features = datasets.Features(
+                {
+                    "document_id": datasets.Value("string"),
+                    "text": datasets.Value("string"),
+                    "textual": [datasets.Value("string")],
+                    "intuitive": [datasets.Value("string")],
+                }
+            )
+
+        elif self.config.schema == "bigbio_text":
+            features = schemas.text_features
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager: datasets.DownloadManager) -> List[datasets.SplitGenerator]:
+        """Returns SplitGenerators."""
+
+        if self.config.data_dir is None:
+            raise ValueError("This is a local dataset. Please pass the data_dir kwarg to load_dataset.")
+        else:
+            data_dir = self.config.data_dir
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={
+                    "data_dir": data_dir,
+                    "split": "train",
+                },
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.TEST,
+                gen_kwargs={
+                    "data_dir": data_dir,
+                    "split": "test",
+                },
+            ),
+        ]
+
+    @staticmethod
+    def _get_source_sample(sample):
+        textual_labels = [
+            disease_name + "_" + sample["textual"][disease_name]
+            for disease_name in sample["textual"].keys()
+            if sample["textual"][disease_name]
+        ]
+        intuitive_labels = [
+            disease_name + "_" + sample["intuitive"][disease_name]
+            for disease_name in sample["intuitive"].keys()
+            if sample["intuitive"][disease_name]
+        ]
+        return {
+            "document_id": sample["document_id"],
+            "text": sample["text"],
+            "textual": textual_labels,
+            "intuitive": intuitive_labels,
+        }
+
+    @staticmethod
+    def _get_bigbio_sample(sample):
+        labels = [
+            task + "_" + disease_name + "_" + sample[task][disease_name]
+            for task in ["textual", "intuitive"]
```

Comment:
just wanting to double check that the sep `_` doesn't appear in any of `[task, disease_name, sample[task][disease_name]` i.e. that someone would be able to split on `_` and always get 3 elements.

Reply:
Pretty sure that's the case. I posted this on discord yesterday. I wasn't sure what's the best way of formatting this multi-label task that has two annotation types. I will try your second suggestion now, w/o adding the names param since not all samples have e.g. "intuitive" annotation or all disease labels.